pkg/api/compact/v1.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package v1
+
+import (
```

Comment:
At this point I would really be careful - it would be nice to have sort of apis like:
* query APIs + UI
* Rules APIs + UI
* Base APIs + UI (flags, metrics endpoints etc)
* Blocks APIs. 

Naming this compact is rather specifying WHO is using this not WHAT this package gives. I think it should be the latter :hugs: 
Then we can specify WHO is using this in each component by registering those which are needed

Reply:
I agree, before the separate package, I wrote the API as a part of the compact package, and then during the refactor it just slipped my mind. I'd rename it as blocks API.